docs/s3-usage-example.md

```diff
@@ -0,0 +1,69 @@
+# S3 Support in Containerlab
+
+Containerlab supports using S3 URLs to retrieve topology files and startup configurations for network devices.
+
+## Prerequisites
+
+AWS credentials are automatically discovered using the standard AWS credential chain (in order of precedence):
+
+1. **Environment variables** (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`)
+2. **Shared credentials file** (`~/.aws/credentials`)
+3. **IAM roles** (EC2 instance profiles)
+
+Optional: You can also use a `.env` file in the current directory to set credentials.
+
+## Usage Examples
+
+### Topology Files from S3
+
+Deploy a lab using a topology file stored in S3:
+
+```bash
+containerlab deploy -t s3://my-bucket/topologies/my-lab.clab.yml
+```
+
```

Comment:
would you be able to create a bucket with random credentials (or public) so that we can add an integration test to check it out for a simple topology with alpine nodes?